include/cutlass/gemm/kernel/xe_gemm.hpp

```diff
@@ -186,15 +193,12 @@ class GemmUniversal<
 
   static dim3
   get_grid_shape(Params const& params) {
-    int batch_count = 1;
-    if constexpr (cute::rank(ProblemShape{}) == 4) {
-      batch_count = cute::size<3>(params.problem_shape);
+    dim3 grid = TileScheduler::get_tiled_cta_shape_mnl(params.problem_shape, TileShape{}, ClusterShape{});
+    if(params.scheduler.raster_order_ == TileScheduler::RasterOrder::AlongN) {
+      return {grid.y, grid.x, grid.z};
+    } else {
+      return {grid.x, grid.y, grid.z};
     }
```

Comment:
I think we can remove this `if` condition since we agreed not to have index switching.

```suggestion
    return dim3{grid.y, grid.x, grid.z};
```